api/src/test/java/org/openmrs/validator/VisitValidatorTest.java

```diff
@@ -243,209 +243,241 @@ public void validate_shouldFailIfAnAttributeIsBad() {
 		new VisitValidator().validate(visit, errors);
-		
+
 		assertFalse(errors.hasFieldErrors("stopDatetime"));
 	}
-	
+
 	private boolean patientHasVoidedVisit(Patient patient, long startInMillis, long stopInMillis) {
-		
+
 		// To get voided visit from the past, both inactive AND voided visits are queried.
 		//
 		List<Visit> visitList = Context.getVisitService().getVisitsByPatient(patient, true, true);
 		for (Visit visit : visitList) {
 			if (visit.getStartDatetime() != null && visit.getStartDatetime().getTime() == startInMillis
-			        && visit.getStopDatetime() != null && visit.getStopDatetime().getTime() == stopInMillis
-			        && visit.getVoided()) {
+					&& visit.getStopDatetime() != null && visit.getStopDatetime().getTime() == stopInMillis
+					&& visit.getVoided()) {
 				return true;
 			}
 		}
 		return false;
 	}
-	
+
 	/**
-	 * @see VisitValidator#validate(Object,Errors)
+	 * @see VisitValidator#validate(Object, Errors)
 	 */
 	@Test
 	public void validate_shouldPassValidationIfFieldLengthsAreCorrect() {
 		Visit visit = makeVisit(42);
 		visit.setVoidReason("voidReason");
-		
+
 		Errors errors = new BindException(visit, "visit");
 		new VisitValidator().validate(visit, errors);
 		assertEquals(false, errors.hasFieldErrors("voidReason"));
 	}
-	
+
 	/**
-	 * @see VisitValidator#validate(Object,Errors)
+	 * @see VisitValidator#validate(Object, Errors)
 	 */
 	@Test
 	public void validate_shouldFailValidationIfFieldLengthsAreNotCorrect() {
 		Visit visit = makeVisit(42);
 		visit
-		        .setVoidReason("too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text");
-		
+				.setVoidReason(
+						"too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text too long text");
+
 		Errors errors = new BindException(visit, "visit");
 		new VisitValidator().validate(visit, errors);
 		assertEquals(true, errors.hasFieldErrors("voidReason"));
 	}
+
+	@Test
+	public void validate_shouldFailIfStartDateTimeIsBeforeBirthDate() {
+		Calendar calendar = Calendar.getInstance();
+		Visit visit = new Visit();
+		visit.setVisitType(visitService.getVisitType(1));
+		visit.setPatient(Context.getPatientService().getPatient(2));
+		visit.setStartDatetime(calendar.getTime());
+		Patient p = visit.getPatient();
+		if (p.getBirthdate().after(visit.getStartDatetime()) && !p.getBirthdateEstimated()) {
+			Errors errors = new BindException(visit, "visit");
+			new VisitValidator().validate(visit, errors);
+			assertTrue (errors.hasFieldErrors("startDateTime"));
+		}
+
+		if (p.getBirthdateEstimated()){
+			if (p.getAge()> 2) {
```

Comment:
Is it possible for me to have just one if clause in the test to check if the birthdate is after a visit date for the said patient? Then I make another test which handles the estimated birthdate situation